src/Tizen.NUI/src/public/WindowSystem/QuickPanelService.cs

```diff
@@ -0,0 +1,266 @@
+        public enum QuickPanelType
+        {
+            /// <summary>
+            /// Unknown type. There is no quickpanel service.
+            /// </summary>
+            Unknown = 0x0,
+            /// <summary>
+            /// System default type.
+            /// </summary>
+            SystemDefault = 0x1,
+            /// <summary>
+            /// Context menu type.
+            /// </summary>
+            ContextMenu = 0x2,
+            /// <summary>
+            /// Apps menu type.
+            /// </summary>
+            AppsMenu = 0x3,
+        }
+
+        /// <summary>
+        /// Effect type.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public enum EffectType
+        {
+            /// <summary>
+            /// Swipe effect
+            /// </summary>
+            Swipe = 0,
+            /// <summary>
+            /// Move effect
+            /// </summary>
+            Move = 1,
+            /// <summary>
+            /// App control effect
+            /// </summary>
+            Custom = 2,
+        }
+
+        /// <summary>
+        /// Creates a new Quickpanel Service handle.
+        /// </summary>
+        /// <param name="tzShell">The TzShell instance.</param>
+        /// <param name="win">The window to provide service of the quickpanel.</param>
+        /// <param name="type">The type of quickpanel service.</param>
+        /// <exception cref="Tizen.Applications.Exceptions.OutOfMemoryException">Thrown when the memory is not enough to allocate.</exception>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <exception cref="ArgumentNullException">Thrown when a argument is null.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public QuickPanelService(TizenShell tzShell, Window win, QuickPanelType type)
+        {
+            if (tzShell == null)
+            {
+                throw new ArgumentNullException(nameof(tzShell));
+            }
+            if (tzShell.GetNativeHandle() == IntPtr.Zero)
+            {
+                throw new ArgumentException("tzShell is not initialized.");
+            }
+            if (win == null)
+            {
+                throw new ArgumentNullException(nameof(win));
+            }
+
+            _tzsh = tzShell;
+            _tzshWin = win.GetNativeId();
+            _tzshQpService = Interop.QuickPanelService.CreateWithType(_tzsh.GetNativeHandle(), (IntPtr)_tzshWin, (int)type);
+            if (_tzshQpService == IntPtr.Zero)
+            {
+                int err = Tizen.Internals.Errors.ErrorFacts.GetLastResult();
+                _tzsh.ErrorCodeThrow(err);
+            }
+        }
+
+        /// <summary>
+        /// Destructor.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when failed of invalid argument.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        ~QuickPanelService()
```

Comment:
could you check the following NUI dispose related codes?

https://github.com/Samsung/TizenFX/blob/master/src/Tizen.NUI/src/public/BaseHandle.cs#L114
https://github.com/Samsung/TizenFX/blob/master/src/Tizen.NUI/src/public/BaseHandle.cs#L300
https://github.com/Samsung/TizenFX/blob/master/src/Tizen.NUI/src/public/BaseHandle.cs#L453

nui has the DisposeQueue so if user doesn't call the Dispose() explicitly, the target object is added in DisposeQueue to be dipsposed internally.

please check. ^^;

Reply:
I changed destructor and dispose all of classes.
And I change Exception in Dispose to ArgumentException to MemberAccessException.
Because in Destroy function return error only NativeHandle is invalid.